server/webapp/WEB-INF/rails.new/spec/presenters/api_v2/config/environment_config_representer_spec.rb

```diff
@@ -0,0 +1,75 @@
+##########################################################################
+# Copyright 2017 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+require 'spec_helper'
+
+describe ApiV2::Config::EnvironmentConfigRepresenter do
+
+  describe :serialize do
+    it 'renders an environment with hal representation' do
+      presenter = ApiV2::Config::EnvironmentConfigRepresenter.new(get_environment_config)
+      actual_json = presenter.to_hash(url_builder: UrlBuilder.new)
+
+      expect(actual_json).to have_links(:self, :find, :doc)
+      expect(actual_json).to have_link(:find).with_url('http://test.host/api/admin/environments/:environment_name')
+      expect(actual_json).to have_link(:self).with_url('http://test.host/api/admin/environments/dev')
+      expect(actual_json).to have_link(:doc).with_url('https://api.gocd.io/#environment-config')
+
+      actual_json.delete(:_links)
+      expect(actual_json).to eq({name: 'dev',
+                                 pipelines: [ApiV2::Config::PipelineConfigSummaryRepresenter.new(com.thoughtworks.go.config.EnvironmentPipelineConfig.new('dev-pipeline')).to_hash(url_builder: UrlBuilder.new)],
+                                 agents: [ApiV2::AgentSummaryRepresenter.new(EnvironmentAgentConfig.new('dev-agent')).to_hash(url_builder: UrlBuilder.new),
+                                          ApiV2::AgentSummaryRepresenter.new(EnvironmentAgentConfig.new('omnipresent-agent')).to_hash(url_builder: UrlBuilder.new)],
+                                 environment_variables: [{:secure => false, :name => 'username', :value => 'admin'}]})
```

Comment:
Make use of the EnvironmentVariableConfigRepresenter to represent the env vars present in an environment.